tests/datasets/test_streaming.py

```diff
@@ -34,17 +44,24 @@ def get_fake_samples_decoders(num_samples: int) -> Tuple[List[Dict[str, bytes]],
     return samples, decoders
 
 
-def write_synthetic_streaming_dataset(dirname: str, samples: List[Dict[str, bytes]], shard_size_limit: int) -> None:
+def write_synthetic_streaming_dataset(dirname: str,
+                                      samples: List[Dict[str, bytes]],
+                                      shard_size_limit: int,
+                                      compression: Optional[str] = StreamingDatasetWriter.default_compression) -> None:
     first_sample_fields = list(samples[0].keys())
-    with StreamingDatasetWriter(dirname=dirname, fields=first_sample_fields,
-                                shard_size_limit=shard_size_limit) as writer:
+    with StreamingDatasetWriter(dirname=dirname,
+                                fields=first_sample_fields,
+                                shard_size_limit=shard_size_limit,
+                                compression=compression) as writer:
         writer.write_samples(samples=samples)
 
 
 @pytest.mark.timeout(10)
 @pytest.mark.parametrize('num_samples', [100, 10000])
 @pytest.mark.parametrize('shard_size_limit', [1 << 8, 1 << 16, 1 << 24])
-def test_writer(remote_local: Tuple[str, str], num_samples: int, shard_size_limit: int) -> None:
+@pytest.mark.parametrize('compression', ['gz', None])
```

Comment:
Can you remove this parameterization here? This will 2x the number of tests across num_samples and shard_size_limit. I believe the writer is tested below.